pkg/rebuild/maven/gradleinfer.go

```diff
@@ -51,17 +55,48 @@ func GradleInfer(ctx context.Context, t rebuild.Target, mux rebuild.RegistryMux,
 	if err != nil {
 		return nil, errors.Wrap(err, "fetching JDK")
 	}
-
+	// Check if gradlew is present in the commit
+	var systemGradle string
+	hasGradleWrapper, err := isGradleWrapperPresent(commitObject)
+	if err != nil {
+		return nil, errors.Wrap(err, "checking for gradle wrapper")
+	}
+	if !hasGradleWrapper {
+		systemGradle = gradleVersion
+		log.Printf("Gradle wrapper (gradlew) not found in the repository. Will use system-installed Gradle.")
+	}
 	return &GradleBuild{
 		Location: rebuild.Location{
 			Repo: repoConfig.URI,
 			Dir:  buildGradleDir,
 			Ref:  ref,
 		},
-		JDKVersion: jdk,
+		JDKVersion:   jdk,
+		SystemGradle: systemGradle,
 	}, nil
 }
 
+func isGradleWrapperPresent(commit *object.Commit) (bool, error) {
```

Comment:
We should be able to simplify this to just check the one path we're interested in with Tree.File(). No need to loop over the whole Tree.